src/Conversion/ONNXToTOSA/NN/MaxPoolSingleOut.cpp

```diff
@@ -25,6 +25,44 @@ namespace onnx_mlir {
 
 namespace {
 
+// This calculates the values that need to be added to the padding in order to
+// simulate the ceil mode
+llvm::SmallVector<int64_t> getCeilConstants(int64_t ceil,
+    llvm::ArrayRef<int64_t> inputShape, mlir::ArrayAttr kernelShape,
+    mlir::ArrayAttr strides, mlir::ArrayAttr pads) {
+  // This avoids having more if statements when creating the padding const.
+  if (!ceil) {
+    return llvm::SmallVector<int64_t>{0, 0};
+  }
+  SmallVector<int64_t> kernelShapeVec = extractFromI64ArrayAttr(kernelShape);
+
+  // Stride and pad are optional attributes. If they are non-existing, we get
+  // the standard values.
+  SmallVector<int64_t> stridesVec;
+  if (strides) {
+    stridesVec = extractFromI64ArrayAttr(strides);
+  } else {
+    stridesVec = {1, 1};
+  }
```

Comment:
You can set the value of stridesVec to {1,1} by default and only modify it if `strides` is set (thus getting rid of the else clause)